src/components/NewClusterModal.js

```diff
@@ -134,13 +134,22 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
     onSuccess: PropTypes.func.isRequired
   }
 
+  makeWorkspaceObj() {
+    const { namespace, name } = this.props
+    return { workspace: { namespace, name } }
+  }
+
   constructor(props) {
     super(props)
     const currentCluster = this.getCurrentCluster()
     const { cloudService, ...currentConfig } = normalizeRuntimeConfig(currentCluster?.runtimeConfig || { masterMachineType: 'n1-standard-4' })
     const { numberOfWorkers } = currentConfig // want these to be put into state below, unlike cloudService
     const currentPersistentDisk = this.getCurrentPersistentDisk()
 
+    Ajax().Metrics.captureEvent(Events.cloudEnvironmentConfigOpen, {
```

Comment:
This produces a side effect, so it should go in `componentDidMount` instead of the constructor.